app/models/spree/order_decorator.rb

```diff
@@ -227,6 +227,22 @@ def total_tax
     (adjustments + price_adjustments).sum &:included_tax
   end
 
+  def tax_adjustments
+    adjustments.with_tax +
+      line_items.includes(:tax_adjustments).map {|li| li.tax_adjustments}.flatten
+  end
+
+  def tax_adjustment_totals
+    Hash[tax_adjustments.group_by(&:label).map do |label, adjustments|
+      [adjustments.first.originator, adjustments.sum(&:amount)]
```

Comment:
Ok, so there is a problem here, not your fault, it's because the way tax is stored is not particularly well designed...

`tax_adjustments` are all adjustments with a tax component. The adjustment `amount` may be all tax (eg. `originator_type: Spree::TaxRate`), or only partially tax (eg. `originator_type: EnterpriseFee` or `originator_type: Spree::ShippingMethod`). This method is treating them as though they are all `originator_type: Tax::Rate` adjustments. 

The problem is that even if we know the type of the originator, the way we currently calculate and store the tax component is different for each type of adjustable: EnterpriseFee adjustments *may* have a Spree::TaxRate, but it is not hard linked, because if the adjustment is on a LineItem, it may be inherited from a Product. ShippingMethod adjustments are completely different again, because they don't have a TaxRate at all, the rate is calculated via a system-wide percentage applied to shipments...

This is why we need to upgrade Spree... 😞